cylc/flow/scripts/validate.py

```diff
@@ -98,7 +95,6 @@
         dest="profile_mode",
         sources={'validate'}
     ),
-    VALIDATE_RUN_MODE,
```

Comment:
Why has this been removed?

Reply:
Because the choices available in this option are `    choices=list(WORKFLOW_RUN_MODES),`, and the only checking done by `run_modes.nonlive.run_mode_validate_checks` are for skip mode, as set by the `run mode` config key.

I guess that you could have an apparently similar option which forces checking of `[[[skip]]]` mode sections as if you were broadcasting `run mode = skip` later, but this feels like overkill.

Perhaps a little refactoring is required, however, because I think that this highlights some nonfunctional code in my PR. :)

Reply:
```cylc
[scheduling]
    [[graph]]
        R1 = a
[runtime]
    [[a]]
        [[[simulation]]]
            fail cycle points = invalid
```
```
$ cylc validate
Valid for cylc-8.4.0.dev
```

Reply:
Yes, that's wrong. Willfix

Reply:
This is a skip mode PR, which shouldn't alter simulation mode functionality. So that's out of scope. 

Skip would be, but `cylc validate --mode skip` isn't.

Reply:
You added a test for the current behaviour here: https://github.com/cylc/cylc-flow/pull/6213/files#diff-91a7012680f87b1a859a22f8bf615342391a6c3ddacbdb24ae4bcc9470ee5727 (and removed it in this PR!)

Also see https://github.com/cylc/cylc-flow/pull/6039#issuecomment-2227848490

> Add skip mode to test for `cylc validate --mode=sim` similar to the tests in https://github.com/cylc/cylc-flow/pull/6213

(Not sure what you meant by this though)

Reply:
> (Not sure what you meant by this though)

At one point this PR would have allowed you to do `cylc validate --mode=sim`, but I removed this feature after a discussion with Oliver because it was making things needlessly complex.

Reply:
> At one point this PR would have allowed you to do `cylc validate --mode=sim`

Do you mean `cylc validate --mode=skip`? You could already do `cylc validate --mode=simulation` before this PR

Reply:
> On this branch, there is now no way to validate `[runtime][X][simulation]fail cycle points` other than running the workflow.

I'm happy to push this (new) problem into a follow-up issue

Reply:
I think I've fixed, will push soon.

Reply:
What's the status on this fix? I still can't validate `[runtime][X][simulation]fail cycle points` as of a67855aaa3749ed514da0d4ee8e6ccd1d024375d

Reply:
I thought that I'd fixed it, but I haven't. Will add a test.